src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -218,7 +219,7 @@ private static void addAwsConfigGroup(final ConfigDef configDef) {
             AWS_S3_BUCKET_NAME_CONFIG,
             Type.STRING,
             null,
-            new ConfigDef.NonEmptyString(),
+            new BucketNameValidator(),
```

Comment:
Why we are validating here and not in getter since there seems to be 3 ways to provide the bucket name?

Reply:
There are only 2 ways (1 deprecated and the current config) as far as I can see; and if you are referring to `getAwsS3BucketName`, it may be too late to perform the name validation as config may already be accepted.

Reply:
Oh, yes, you are right + I misread the method thinking that we are trying to get the value using 3 names, not too. Should we reuse the validator instead of anonymous class in the deprecated property definition?

Reply:
Was trying to keep it as before, to include the logging warning. We can get rid of it once deprecated configs are removed at some point (🤞🏽)